src/Microsoft.TestPlatform.PlatformAbstractions/net451/System/ProcessHelper.cs

```diff
@@ -15,21 +17,98 @@ namespace Microsoft.VisualStudio.TestPlatform.PlatformAbstractions;
 
 public partial class ProcessHelper : IProcessHelper
 {
+    [DllImport("kernel32.dll", SetLastError = true)]
+    private static extern bool IsWow64Process2(IntPtr process, out ushort processMachine, out ushort nativeMachine);
+
+    private const ushort IMAGE_FILE_MACHINE_ARM64 = 0xAA64;
+    private const ushort IMAGE_FILE_MACHINE_UNKNOWN = 0;
+
     /// <inheritdoc/>
     public string GetCurrentProcessLocation()
-    {
-        return Path.GetDirectoryName(GetCurrentProcessFileName());
-    }
+        => Path.GetDirectoryName(GetCurrentProcessFileName());
 
+    /// <inheritdoc/>
     public IntPtr GetProcessHandle(int processId)
-    {
-        return Process.GetProcessById(processId).Handle;
-    }
+        => Process.GetProcessById(processId).Handle;
 
     /// <inheritdoc/>
     public PlatformArchitecture GetCurrentProcessArchitecture()
+        =>
+        IntPtr.Size == 8
+        ? IsArm64()
+            ? PlatformArchitecture.ARM64
+            : PlatformArchitecture.X64
+        : PlatformArchitecture.X86;
+
+    private static bool IsArm64()
     {
-        return IntPtr.Size == 8 ? PlatformArchitecture.X64 : PlatformArchitecture.X86;
+        try
+        {
+            var currentProcess = Process.GetCurrentProcess();
+            if (!IsWow64Process2(currentProcess.Handle, out ushort processMachine, out ushort nativeMachine))
+                throw new Win32Exception();
```

Comment:
Could we maybe add a message to the exception or at least add some log before so that we have some info about what went wrong?